tools/prepare_software_list.py

```diff
@@ -0,0 +1,290 @@
+        return False
+    now = datetime.utcnow()
+    if now - renewed > ENTRY_LIFETIME:
+        print('Client entry expired for', name)
+        return False
+    return True
+
+
+def parse_doap_infos(doap_file: str
+                     ) -> Optional[dict[str, Union[str, list[str], None]]]:
+    '''Parse DOAP file and return infos'''
+    try:
+        doap = ET.parse(
+            DOWNLOAD_PATH / f'doap_files/{doap_file}.doap')
+    except FileNotFoundError:
+        return None
+
+    info: dict[str, Union[str, list[str], None]] = {}
+
+    info['name'] = None
+    doap_name = doap.find(DOAP_NAME)
+    if doap_name is not None:
+        info['name'] = doap_name.text
+
+    info['shortdesc'] = None
+    doap_shortdesc = doap.find(DOAP_SHORTDESC)
+    if doap_shortdesc is not None:
+        info['shortdesc'] = doap_shortdesc.text
+
+    info['os'] = []
+    for entry in doap.findall(DOAP_OS):
+        info['os'].append(entry.text)
+    if not info['os']:
+        info['os'] = ['Other']
+
+    info['logo'] = None
+    doap_logo = doap.find(DOAP_LOGO)
+    if doap_logo is not None:
+        info['logo'] = doap_logo.attrib[RDF_RESOURCE]
+
+    return info
+
+
+def check_file_size(file_path: Path) -> None:
+    '''
+    Check if file size is greater than 300 KiB and if so, resize image
+    '''
+    try:
+        file_size = os.path.getsize(file_path)
+    except OSError as error:
+        print('An error occurred while trying to open logo:', error)
+    if file_size > 300000:
+        try:
+            with Image.open(file_path) as img:
+                width, height = img.size
+                new_width = 400
+                new_height = int(new_width * height / width)
+                img = img.resize(
+                    (int(new_width), int(new_height)), Resampling.LANCZOS)
+                img.save(file_path)
+                print(f'Logo at {file_path} '
+                      f'(file size: {file_size}) has been resized')
+        except OSError as error:
+            print('An error occurred while trying to resize logo:', error)
+
+
+def process_logo(client_name: str, uri: str) -> Optional[str]:
+    '''
+    Download client logo and return logo URI
+    '''
+    image_url = urlparse(uri)
+    _, extension = os.path.splitext(image_url.path)
+    file_name = f'{client_name}{extension}'
+    success = download_file(
+        uri,
+        Path(file_name))
+    if not success:
+        return None
+
+    if extension[1:] in ('png', 'jpg'):
+        check_file_size(DOWNLOAD_PATH / file_name)
+
+    logo_uri = f'/images/clients/{client_name}{extension}'
+    shutil.copyfile(
+        DOWNLOAD_PATH / file_name,
+        Path(LOGOS_PATH / file_name))
+    return logo_uri
+
+
+def prepare_clients_list() -> None:
+    '''
+    Download and prepare clients data
+    '''
+    success = download_file(
+        XSF_CLIENTS_LIST_URL, Path('xsf_clients_list.json'))
+    if not success:
+        sys.exit()
```

Comment:
[If I read `sys.exit`](https://docs.python.org/3/library/sys.html#sys.exit) correctly, this exits without any message and without error code. Is that truly what you want?

I would've expected an exception raised here, or an error printed and exit with non-zero exit code, because this effectively makes this thing not do what it's supposed to do.

Reply:
Since `clients.json` is always available locally, this code isn't necessary anymore.